public_html/wp-content/mu-plugins/blocks/includes/components/featured-image.php

```diff
@@ -28,7 +28,7 @@ function render_featured_image( $post, $width, $class_names = [], $image_link =
 	$size         = array( $width, $height );
 
 	$container_classes = array_merge(
-		[ 'wordcamp-image-container', 'wordcamp-featured-image-container' ],
+		[ 'wordcamp-image__featured-image-container' ],
```

Comment:
🤔 In this case, `featured-image` component is actually part of `image` folder, which has its own `index.js`. Guidelines also states that:

> Notably, if your folder contains multiple files, each with their own default exported component, only the element rendered by that of `index.js` can be considered the root.

So that's why I am using `image` as root descriptor and `featured-image` as component. What do you think?

Reply:
Ah I see, so in `php` we have `featured-image` in as the folder name, but in `js` we have `image` as the folder name. What do you think we should give preference to? Or should we rearrange the folders in `php` to reflect `js`?

Reply:
Oh, you're right, good point.

🤔 , wonder if we should really have all those inside an image folder? It makes sense in a way, but also feels kind of awkward to have the extra level there, especially with the way it affects the class names. This might be simpler:

```
components
 |-- avatar
 |-- featured-image
 |-- image-inspector-controls
```

Although, [this comment](https://github.com/WordPress/wordcamp.org/issues/96#issuecomment-491443218) might be relevant here too:

> For us the block/ and component/ folders are just transparent organizational devices for the filesystem, rather than part of the actual component hierarchy in the DOM. Maybe that's a useful way to think about it?

Even if we keep the `image` container folder, I'm not sure it necessarily needs to be taken into account in the class names.

We could look at what G does for some conventions, IIRC there are some nested components in a few places too, but it also feels like generally the hierarchy is kept mostly flat.

----

If we were to make the PHP consistent w/ the JS, I guess we'd want it structured more like this:

* `includes/components/featured-image/controller.php`
* `includes/components/featured-image/view.php`

...rather than being split between `/includes` and `/views. That feels kinda nice, because it reflects the component-ized nature of the system.

Another discrepancy is that the views for the PHP components are embedded in the controller files, while the views for the PHP blocks are separated into the `views` folder. I'm guessing that was just done because they're fairly small, but it might be nice to separate them for consistency, readability, and separation-of-concerns.

Reply:
🤔 

Or, I wonder if we should instead just move the PHP files into `source`, instead of keeping them in a separate folder? That way we'd have:

```
source
|-- blocks
|-- |-- organizers
|-- |-- |-- controller.php
|-- |-- |-- edit.js
|-- |-- |-- index.js
|-- |-- |-- view.php
|-- |-- |-- ...etc
```

Reply:
We agreed on suggestion by @iandunn that it makes more logical sense in this case to move `php` files to source folder as well.

Reply:
Done in https://github.com/WordPress/wordcamp.org/pull/123/commits/909a6ad397da18e573c1584e746cf459a3eff99d